ex3/client/main.cpp

```diff
@@ -5,36 +5,53 @@
 #include "vector"
 #include "../utils.h"
 #include "TCPClient.h"
+#include "thread"
+#include "queue"
+#include "mutex"
+#include "condition_variable"
+
+std::condition_variable cv;
+std::mutex mutex;
+
+void receiving(std::queue<std::string> &receiving_queue, TCPClient client) {
+    while (true) {
+        std::string msg = client.recv();
+        receiving_queue.push(msg);
+        cv.notify_one();
+    }
+}
+
+std::string receiveFromServer(std::queue<std::string> &receiving_queue) {
+    std::unique_lock<std::mutex> uniqueLock(mutex);
+    cv.wait(uniqueLock, [&receiving_queue] { return !(receiving_queue.empty()); });
+    std::string res = receiving_queue.front();
+    receiving_queue.pop();
+    return res;
+}
+
 
 int main() {
     //Crete clients
     TCPClient client(inet_addr("127.0.0.1"), htons(55555));
     //Get parameters from user
     std::string parameters;
-    std::getline(std::cin, parameters);
-    std::vector<std::string> params = utils::split(parameters, ' ');
+    std::queue<std::string> msgQueue;
+    std::thread receivingThread(receiving, std::ref(msgQueue), client);
+    receivingThread.detach();
     //Read the iris data from the csv file
-    std::vector<std::vector<std::string>> unclassified = utils::readCSV(params[0]);
-    //Set the client according to the request
-    //Split the data in order to send it
-    std::string msg;
-    for (const std::vector<std::string> &i: unclassified) {
-        for (const std::string &j: i) {
-            msg.append(j);
-            msg.append(" ");
+    while (true) {
+        std::getline(std::cin, parameters);
+        std::vector<std::string> params = utils::split(parameters, ' ');
+        if (params[0] == "exit") {
+            client.send("exit");
+            break;
         }
-        msg.pop_back();
-        msg.append("\n");
+        std::vector<std::vector<std::string>> unclassified = utils::readCSV(params[0]);
+        std::string msg = utils::joinVector(unclassified);
```

Comment:
We should be able to just read the file.
Replace the space delim with `,` to match the CSV format.

Reply:
Created an issue.